src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java

```diff
@@ -0,0 +1,224 @@
+		when(tunnelingService.getExternalServiceAddresses(order.getId())).thenReturn(null);
+		when(tunnelingService.getExternalServiceAddresses(order.getId())).thenThrow(new RuntimeException("Any Exception"));
+		
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+	}
+	
+	@Test
+	public void testProcesseComputeOrderInstanceInactive() {
+		Order order = createOrder();
+		SynchronizedDoublyLinkedList list = addInSpawningOrderList(order);
+		order = list.getNext();
+		list.resetPointer();
+		
+		OrderInstance orderInstance = new OrderInstance(DEFAULT_ORDER_INSTANCE_ID);
+		order.setOrderInstance(orderInstance);
+		
+		ComputeOrderInstance computeOrderInstance = Mockito.spy(ComputeOrderInstance.class);
+		computeOrderInstance.setState(InstanceState.INACTIVE);
+		Mockito.when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(computeOrderInstance);
+		
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+
+		Order test = SharedOrderHolders.getInstance().getSpawningOrdersList().getNext();
+		assertNotNull(test);
+		Assert.assertEquals(OrderState.SPAWNING, test.getOrderState());	
+	}
+	
+	@Test
+	public void testProcesseComputeOrderInstanceFailed() {
+		Order order = createOrder();
+		SynchronizedDoublyLinkedList list = addInSpawningOrderList(order);
+		order = list.getNext();
+		
+		OrderInstance orderInstance = new OrderInstance(DEFAULT_ORDER_INSTANCE_ID);
+		order.setOrderInstance(orderInstance);
+		
+		ComputeOrderInstance computeOrderInstance = Mockito.spy(ComputeOrderInstance.class);
+		computeOrderInstance.setState(InstanceState.FAILED);
+		Mockito.when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(computeOrderInstance);
+		
+		Assert.assertNull(SharedOrderHolders.getInstance().getFailedOrdersList().getNext());
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+		Assert.assertNull(SharedOrderHolders.getInstance().getSpawningOrdersList().getNext());
+		
+		Order test = SharedOrderHolders.getInstance().getFailedOrdersList().getNext();		
+		Assert.assertEquals(order.getOrderInstance(), test.getOrderInstance());
+		Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+	}
+	
+	@Test 
+	public void testProcesseComputeOrderInstanceConfiguredToThrowException() {
+		Order order = this.createOrder();
+		order.setOrderState(OrderState.SPAWNING);
+		OrderInstance orderInstance = order.getOrderInstance();
+		order.setOrderInstance(orderInstance);
+		when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(null);
+		when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenThrow(new RuntimeException("Any Exception"));
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+	}
+	
+	@Test
+	public void  testMethodProcessSpawningOrderConfiguredToThrowException() {
```

Comment:
What is the main goal of this test case? There is any chance of the getOrderInstance throw an Exception?